fs/dafsserver/dafsserver.cpp

```diff
@@ -2056,8 +2057,16 @@ class CRemoteIndexReadActivity : public CRemoteIndexBaseActivity
                 while (keyManager->lookup(true))
                 {
                     const byte *keyRow = keyManager->queryKeyBuffer();
-                    retSz = translator->translate(outBuilder, *this, keyRow);
-                    if (retSz)
+                    if (translator)
+                        retSz = translator->translate(outBuilder, *this, keyRow);
+                    else
+                    {
+                        outBuilder.ensureRow();
+                        retSz = keyManager->queryRowSize();
+                        memcpy(outBuilder.getSelf(), keyRow, retSz);
+                    }
+
+                    if (retSz) // JCSMORE is this poss.?
```

Comment:
@ghalliday - are there any circumstacnes where this could be 0?

Reply:
I don't think so - same as the situation with the disk read, transform() cannot filter.

Reply:
ok I'll change it to a dbgassertex as in other cases.